src/main/java/org/apache/maven/resolver/internal/ant/AntRepoSys.java

```diff
@@ -821,17 +821,13 @@ private void populateCollectRequest( CollectRequest collectRequest, Task task, R
                     {
                         continue;
                     }
-                    Dependency dependency = new Dependency();
+                    final Dependency dependency = new Dependency();
                     dependency.setCoords( line );
                     dependencies.add( dependency );
                 }
             }
-            finally
-            {
-                reader.close();
-            }
         }
-        catch ( IOException e )
+        catch ( final IOException e )
```

Comment:
Why this change? The ticket describes changes related to try-with-resource?

Reply:
HI @arturobernalg 
Correct. The main change it's use try-with-resource. Just a improve of the same code

Reply:
That's simply not related and shouldn't being done with the same issue. I would suggest to focus on the subject of the issue...

Reply:
Reverted all the secondary micro changes.
TY